src/object_schema.cpp

```diff
@@ -107,11 +108,31 @@ Property *ObjectSchema::property_for_name(StringData name) {
     return nullptr;
 }
 
-const Property *ObjectSchema::property_for_name(StringData name) const {
+Property *ObjectSchema::property_for_alias(StringData alias)
+{
+    for (auto& prop : persisted_properties) {
+        if (StringData(prop.alias.empty() ? prop.name : prop.alias) == alias)
```

Comment:
Shouldn't we check for equality on both the name and the alias? Currently if an alias is set on a property, but the actual name is passed, a match won't be found.

`if (StringData(prop.name) == alias || (!prop.alias.empty() && StringData(prop.alias) == alias)` ?